python/lolopy/learners.py

```diff
@@ -137,39 +125,52 @@ def predict(self, X, return_std=False):
         # Get the PredictionResult
         pred_result = self.model_.transform(X_java)
 
+        # Unlink the run data, which is no longer needed (to save memory)
+        self.gateway.detach(X_java)
+
         # Pull out the expected values
-        exp_values = pred_result.getExpected()
-        y_pred = [exp_values.apply(i) for i in range(len(X))]
+        y_pred = self.gateway.jvm.io.citrine.lolo.util.LoloPyDataLoader.getRegressionExpected(pred_result)
```

Comment:
Could you pick a different name to avoid `x = f(x)` type patterns.  Maybe `y_pred_byte`?